tests/shared/src/test/scala/scala/meta/tests/parsers/DeclSuite.scala

```diff
@@ -90,6 +90,33 @@ class DeclSuite extends ParseSuite {
     ) = templStat("type F[-T]")
   }
 
+  test("opaque type F[T]") {
+    val Decl.Type(
+      List(Mod.Opaque()),
+      Type.Name("F"),
+      List(Type.Param(Nil, Type.Name("T"), Nil, Type.Bounds(None, None), Nil, Nil)),
+      Type.Bounds(None, None)
+    ) = templStat("opaque type F[T]")(dialects.Dotty)
+  }
+
+  test("opaque type F <: Bound = X") {
+    val Decl.Type(
+      List(Mod.Opaque()),
+      Type.Name("F"),
+      List(),
+      Type.Bounds(None, None)
+    ) = templStat("opaque type F <: Bound = X")(dialects.Dotty)
+  }
+
+  test("opaque type F <: A & B") {
+    val Decl.Type(
```

Comment:
I have never been a fan of this style of tests because they fail with cryptic error messages. I prefer to use `assertNoDiff` in the following may
```scala
def checkParsed(original: String, expectedStructure: String) =
  test(original) {
    val obtainedStructure = printStructure(parse(original))
    assertNoDiff(obtainedStructure, expectedStructure)
  }

checkParsed(
  "opaque type F <: A & B",
  """
  Type.Bounds(None, Some(Type.And(Type.Name("A"), Type.Name("B"))))...
  """
)
```

We may need to implement a custom `printStructure` that does basic line-wrapping to avoid large lines. The benefit of this style is that when the tests fail we get a more helpful message with copy-pastable string of the obtained behavior.

Reply:
For historical background, I didn't write any of the current parser tests. They existed in the project before I started contributing in 2016. Given that you're actively working on the parser for Scala 3 support I wanted to bring this up in case you think it might help make you more productive. Feel free to ignore this comment if you're happy with the current style of tests.

Reply:
For new suites I added I used my wrapper:
```
  protected def runTestAssert[T <: Tree](code: String)(expected: T)(implicit parser: String => T) {
    val obtained: T = parser(code)
    try {
      assertEquals(obtained, expected)
    } catch {
      case e: Throwable =>
        println(s"Generated stat: \n ${obtained.structure}")
        throw e
    }
  }
```
It is similar to what you suggest except I would prefer to stick with proper types for now as it eases refactoring and catcher errors earlier.

Fact is I don't like that some objects to compare gets a little big and with scalafmt it gets formatted into multiple lines that doesn't necessarily improve readability but no good workaround so far.

E.g.:
```
    runTestAssert[Stat]("enum Color { case R extends Color }")(
      Defn.Enum(
        Nil,
        pname("Color"),
        Nil,
        ctor,
        tpl(
          List(
            Enum.Case(Nil, tname("R"), Nil, ctor, List(init("Color")))
          )
        )
      )
    )
```

Reply:
Cool, I'm happy with whatever approach works best for you 👍